### DISCUSSION TITLE
How to attach `@doc()` to an `@useAuth()` `Model`?

### DISCUSSION THREAD

**Discussion by charlypoly** [microsoft/typespec#3359] [Q&A] [answered]
Hi 👋🏼 

I've been trying to get the following`securitySchemes` with `description` with Typespec:

```json
{
  "securitySchemes": {
    "bearerAuth": {
      "type": "apiKey",
      "in": "header",
      "name": "x-my-key",
      "description": "A custom explanation on API Key"
    }
  }
}
```

Adding a `@doc(...)` on the line before `useAuth(...)` doesn't seem to do the trick.
Looking at the implementation, it should be possible to attach a `description` to a security scheme by using `@doc()`:
https://github.com/microsoft/typespec/blob/69d6cb78ad7116eb38ee84f41a9ec479906ac952/packages/http/src/decorators.ts#L610

However, I did not find how to properly apply the `@doc()` to get it attached to the `Model`. Am I missing something, or is it a bug?

**Comment by timotheeguerin** [comment, marked as answer]
If you create a custom model for your authentication you can do that [playground example](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwOwoKLyoqIFNvbWUgY3VzdG9tIGRvYyAqLwptb2RlbCBNeUF1dGggaXMgQmFzaWPFDXsKCn0KQHNlcnZpY2UKQHVzZcQXKMYrKQpuYW1lc3BhY2UgRGVtb1PGJ8Q1ICAKfQo%3D&e=%40typespec%2Fopenapi3&options=%7B%22linterRuleSet%22%3A%7B%22extends%22%3A%5B%22%40typespec%2Fhttp%2Fall%22%5D%7D%7D)

```tsp
import "@typespec/http";

using TypeSpec.Http;

/** Some custom doc */
model MyAuth is BasicAuth {

}
@service
@useAuth(MyAuth)
namespace DemoService { 
}
```

**Reply by charlypoly** [reply]
Works! Thank you for the quick answer ⚡
